### DISCUSSION TITLE
[Bug] Incorrect image ratio detected after converting file

### DISCUSSION THREAD

**Discussion by JoaoPaul54** [sn4k3/UVtools#670] [Q&A] [answered]
### System

```Text
UVtools v3.12.0 X64
Operative system: Microsoft Windows 10.0.22621 X64
Processor: Intel(R) Core(TM) i9-9900K CPU @ 3.60GHz
Processor cores: 16
Memory RAM: 19.29 / 31.94 GB
Runtime: win10-x64
Framework: .NET 6.0.14
AvaloniaUI: 0.10.18
OpenCV: 4.6.0

Sreens, resolution, working area, usable area:
1: 2560 x 1440 @ 100% (Primary) (On this)
    WA: 2560 x 1392    UA: 2560 x 1392


Path:       C:\Program Files (x86)\UVtools\
Executable: C:\Program Files (x86)\UVtools\UVtools.exe
Loaded file: Base(repaired).sl1 [Version: 0] [Class: SL1File]
```

### Printer and Slicer

- Printer: Anycubic Photon S
- Slicer:  PrusaSlicer

### Description of the bug

Tried using my own profile and by Installing prusa slicer profile for photon S machine from UVTools.

Attempting to convert a SL1 file from prusa slicer to photon, photons or pws all give the same error when opening the new file.

The error is the following:

It looks like this file was sliced with an incorrect image ratio.
Printing this file may produce a stretched model with wrong proportions or a failed print.
Please go back to Slicer configuration and validate the printer resolution and display size.

Resolution: {Width=2560, Height=1440}
Display: {Width=68.04, Height=120.96}
Ratio: 1:27

By checking the sizes shown in UVTools it is indeed showing some weird values

Original file:
![image](https://user-images.githubusercontent.com/36473226/222293585-22261103-94c5-485a-acf0-ad16699c230b.png)

Converted File:
![image](https://user-images.githubusercontent.com/36473226/222293627-9cd76990-83c5-4554-a9bf-b557b3bc8193.png)

### How to reproduce

Open an SL1 file in UVTools
Convert the file to pws, photon or photons file format (probably others)
Open the new file

**Comment by sn4k3** [comment]
The current UVtools profile are correct. Do you change any values in printer properties? Resolution/Display?

The warning message already told the problem, close inspect resolution and display, a value is flipped.

photon S should be: Resolution: 1440 x 2560 and Display: 68.04 x 120.96.

Use UVtools profile manager to install the correct profile into PrusaSlicer, I do not recommend users to create their own profile from scratch, always use UVtools profiles as base for your machine.

If you had a old profile delete it and reinstall UVtools profile

**Comment by JoaoPaul54** [comment]
> Tried using my own profile and by Installing prusa slicer profile for photon S machine from UVTools.

I did both, the error persisted :/

Also, I might be mistakes but wouldn't print volume e the same as the display?

Anycubic website says:
● Printing volume : 115mm *65mm *165mm (4.52″*2.56″*6.1″)

This might not be correct though (wouldn't surprise me)

I'll try to delete the profile once again and redo the conversion meanwhile.

**Reply by sn4k3** [reply]
> I did both, the error persisted :/

I'm currently out, can't test, but from looking at profiles it looks correct to me.
Go to printer tab -> notes, copy everything and paste here.

> Also, I might be mistakes but wouldn't print volume e the same as the display?

it should be, but most of the times it's rounded or a bit smaller than actual display. Because you should never print with full display and then they give a smaller margin on build volume to be safe on specs. Meaning that the actual display is larger and that is important to slice correct images with correct pixel ratio. PhotonS does not have a field to store display size on file, and so UVtools set that information static, once converted to photons it will show the static value instead of the old value, and if set incorrectly before it can raise these warnings.

**Comment by JoaoPaul54** [comment, marked as answer]
I think I found the issue. aparently prusa slicer does not delete any changes we might have done to any profile even though it was deleted and reinstalled, some values were never changed back for some reason. Probably something regarding autosaves and me using the same name for my old profile as the UVTools did.

I wonder if the profiles naming should have a UVTools_ on the beginning to avoid this issues... unfortunately I don't think I can reproduce it again

For now it seems to be working correctly, or at least the error message didn't appear.

If anyone else gets this issue, I had to rename my previous profile and only then reinstall through UVTools

**Reply by sn4k3** [reply]
That's strange, as soon you delete a profile the file is erased from disk. UVtools copy and replace profiles if installed via the manager.

> I wonder if the profiles naming should have a UVTools_ on the beginning to avoid this issues... unfortunately I don't think I can reproduce it again

For users who want to modify printer profile I always recommend to duplicate the UVtools profile and change from there, to avoid a not wanted replace via the manager by mistake
